packages/labs/src/components/tag-input/tagInput.tsx

```diff
@@ -205,22 +207,26 @@ export class TagInput extends AbstractComponent<ITagInputProps, ITagInputState>
 
     private getNextActiveIndex(direction: number) {
         const { activeIndex } = this.state;
-        const totalItems = this.props.values.length;
-
-        if (activeIndex === NONE && direction < 0) {
-            // nothing active, moving left: select last
-            return totalItems - 1;
-        } else if (activeIndex === NONE && direction > 0) {
-            // nothing active, moving right: do nothing
-            return NONE;
+        if (activeIndex === NONE) {
+            // nothing active & moving left: select last defined value. otherwise select nothing.
+            return direction < 0 ? this.findNextIndex(this.props.values.length, -1) : NONE;
         } else {
             // otherwise, move in direction and clamp to bounds.
             // note that upper bound allows going one beyond last item
             // so focus can move off the right end, into the text input.
-            return Utils.clamp(activeIndex + direction, 0, totalItems);
+            return this.findNextIndex(activeIndex, direction);
         }
     }
 
+    private findNextIndex(startIndex: number, direction: number) {
+        const { values } = this.props;
+        let index = startIndex + direction;
+        while (index > 0 && index < values.length && !values[index]) {
```

Comment:
This while loop is a little tricky at a glance. Can we add a comment like:

```tsx
// move in the specified direction, jumping to the other end when we hit a boundary,
// until we reach a non-disabled element
```

Also, what if all elements are disabled? Are we sure there's no potential for an infinite loop here?